src/main/java/pau/task/TaskList.java

```diff
@@ -0,0 +1,247 @@
+package pau.task;
+
+import pau.exception.DeadlineNoEndException;
+import pau.exception.NoDescException;
+import pau.exception.NoSuchTaskException;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Stores the tasks in an ArrayList;
+ */
+public class TaskList {
+
+    /**
+     * The list where the tasks are stored.
+     */
+    private ArrayList<Task> taskList;
+
+    /**
+     * Constructs a new ArrayList for the tasks.
+     */
+    public TaskList() {
+        this.taskList = new ArrayList<>();
+    }
+
+    /**
+     * Retrieves size of the task list.
+     *
+     * @return Size of the task list.
+     */
+    public int listSize() {
+        return this.taskList.size();
+    }
+
+    /**
+     * Retrieves the task at a certain index in the task list.
+     *
+     * @param index The index of the task to be retrieved.
+     * @return The task at the index.
+     */
+    public Task getTask(int index) {
+        return taskList.get(index);
+    }
+
+    /**
+     * Prints out a list of the tasks the user has.
+     */
+    public void checkList() {
```

Comment:
can consider renaming to `printChecklist` to improve clarity since the expected behavior is to print?